File: plot2.R
# plot2.R
# Q2: Have total Baltimore City PM2.5 emissions decreased from 1999 to 2008?
# Bar chart of total emissions for Baltimore City (fips == "24510").

NEI <- readRDS("ExploratoryDataAnalysis/summarySCC_PM25.rds")

# Filter Baltimore City
baltimore <- subset(NEI, fips == "24510")
baltimore_by_year <- aggregate(Emissions ~ year, data = baltimore, FUN = sum)

# Save to PNG
png("plot2.png", width = 480, height = 480)

barplot(
  baltimore_by_year$Emissions,
  names.arg = baltimore_by_year$year,
  col = "tomato",
  xlab = "Year",
  ylab = "Total PM2.5 Emissions (tons)",
  main = "Total PM2.5 Emissions — Baltimore City, MD (1999–2008)",
  ylim = c(0, max(baltimore_by_year$Emissions) * 1.15)
)

dev.off()

File: plot2.py
#!/usr/bin/env python
"""
Plot 2: Have total Baltimore City PM2.5 emissions decreased from 1999 to 2008?
Bar chart of total emissions for Baltimore City (fips == "24510").
"""

import pyreadr
import pandas as pd
import matplotlib.pyplot as plt

# Read data
NEI = pyreadr.read_r("ExploratoryDataAnalysis/summarySCC_PM25.rds")[None]
NEI["Emissions"] = pd.to_numeric(NEI["Emissions"], errors="coerce")
NEI["year"] = NEI["year"].astype(int)

# Filter Baltimore City
baltimore = NEI[NEI["fips"] == "24510"]
baltimore_by_year = baltimore.groupby("year")["Emissions"].sum()

fig, ax = plt.subplots(figsize=(480/100, 480/100), dpi=100)
ax.bar(baltimore_by_year.index.astype(str), baltimore_by_year.values, color="tomato", width=0.5)
ax.set_xlabel("Year")
ax.set_ylabel("Total PM2.5 Emissions (tons)")
ax.set_title("Total PM2.5 Emissions — Baltimore City, MD (1999–2008)")
for i, (yr, val) in enumerate(baltimore_by_year.items()):
    ax.text(i, val + 20, f"{val:.0f}", ha="center", fontsize=9)
plt.tight_layout()
plt.savefig("plot2.png", dpi=100, bbox_inches="tight")
plt.close()
print("Plot 2 saved as plot2.png")
